server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -3492,7 +3481,7 @@ public void testDoubleDeliveryPrimary() throws IOException {
         if (randomBoolean()) {
             Engine.IndexResult indexResult = engine.index(operation);
             assertLuceneOperations(engine, 1, 0, 0);
-            assertThatIfAssertionEnabled(engine.getNumVersionLookups(), n -> assertThat(n).isEqualTo(1L));
+            assertThatIfAssertionEnabled(engine.getNumVersionLookups(), n -> assertThat(n).isEqualTo(0));
```

Comment:
Is it fine that version lookups decreased to 0?

Reply:
Yes, that's the test bug - it should be `0`, and in fact it was before 815ba73, but it got changed during the refactor and then never picked up because the fixed seed meant that this branch was never run.